sync15-adapter/src/sync.rs

```diff
@@ -15,6 +15,8 @@ use util::ServerTimestamp;
 /// Different stores will produce errors of different types.  To accommodate this, we force them
 /// all to return failure::Error, which we expose as ErrorKind::StoreError.
 pub trait Store {
+    fn collection_name(&self) -> String;
```

Comment:
It might be worth having this return a `&str` (or even a `&'static str`, since it's not exactly like we want to encourage allocating these dynamically), to avoid the cost for the cases where you don't actually need it to be an owned string.

In the future I have an idea on how to make the collection names both more efficient and less error prone, but tbh who knows if that will ever happen.